### ISSUE TITLE
[stable/graylog] unknown field "updateStrategy" in io.k8s.api.core.v1.PodSpec

### ISSUE THREAD

**Issue by ghost** [helm/charts#13938]
**Describe the bug**
Error message:
> Error: validation failed: error validating "": error validating data: ValidationError(StatefulSet.spec.template.spec): unknown field "updateStrategy" in io.k8s.api.core.v1.PodSpec

**Version of Helm and Kubernetes**:
Helm:
> $ helm version
> Client: &version.Version{SemVer:"v2.14.0", GitCommit:"05811b84a3f93603dd6c2fcfe57944dfa7ab7fd0", GitTreeState:"clean"}
> Server: &version.Version{SemVer:"v2.14.0", GitCommit:"05811b84a3f93603dd6c2fcfe57944dfa7ab7fd0", GitTreeState:"clean"}

Kubernetes: EKS 1.11

**Which chart**:
stable/graylog

**What happened**:
When I execute the command :

> $ helm install --namespace "graylog" -n "graylog" --set servers.replicas=3 --set tags.install-mongodb=false --set tags.install-elasticsearch=false --set graylog.mongodb.uri="mongodb://mongodb-mongodb-replicaset-0.mongodb-mongodb-replicaset.graylog.svc.cluster.local:27017/graylog?replicaSet=rs0" --set elasticsearch.hosts=http://elasticsearch-client.graylog.svc.cluster.local:9200 stable/graylog

It returns:

> Error: validation failed: error validating "": error validating data: ValidationError(StatefulSet.spec.template.spec): unknown field "updateStrategy" in io.k8s.api.core.v1.PodSpec

**What you expected to happen**:
No error.

**How to reproduce it** (as minimally and precisely as possible):
When creating Graylog at first time.

**Anything else we need to know**:
NA

**Comment by tru-cstavares** [comment]
Hi, discovered the issue. The key **updateStrategy** is on wrong line in file templates/statefulset.yaml

Your code shows the key **updateStrategy** under spec.template.spec, but it should be under spec as shown bellow:

>       1 apiVersion: apps/v1
>       2 kind: StatefulSet
>       3 metadata:
>       4   name: {{ template "graylog.fullname" . }}
>       5   labels:
>       6     app.kubernetes.io/name: {{ template "graylog.name" . }}
>       7     helm.sh/chart: {{ template "graylog.chart" . }}
>       8     app.kubernetes.io/managed-by: "{{ .Release.Service }}"
>       9     app.kubernetes.io/instance: "{{ .Release.Name }}"
>      10     app.kubernetes.io/version: "{{ .Chart.AppVersion }}"
>      11 spec:
>      12   serviceName: {{ template "graylog.fullname" . }}
>      13   replicas: {{ .Values.graylog.replicas }}
>      14   selector:
>      15     matchLabels:
>      16       app.kubernetes.io/name: {{ template "graylog.name" . }}
>      17       app.kubernetes.io/instance: "{{ .Release.Name }}"
>      18       app.kubernetes.io/managed-by: "{{ .Release.Service }}"
>      19   updateStrategy:
>      20     type: {{ .Values.graylog.updateStrategy }}
>      21   template:
>      22     metadata:
>      23       labels:
>      24         app.kubernetes.io/name: {{ template "graylog.name" . }}
>      25         helm.sh/chart: {{ template "graylog.chart" . }}
>      26         app.kubernetes.io/managed-by: "{{ .Release.Service }}"
>      27         app.kubernetes.io/instance: "{{ .Release.Name }}"
>      28         app.kubernetes.io/version: "{{ .Chart.AppVersion }}"
>      29       annotations:
>      30       {{- if .Values.graylog.podAnnotations }}
>      31       {{- range $key, $value := .Values.graylog.podAnnotations }}
>      32         {{ $key }}: {{ $value | quote }}
>      33       {{- end }}
>      34       {{- end }}
>      35       {{- if .Values.graylog.metrics.enabled }}
>      36         prometheus.io/scrape: "true"
>      37         prometheus.io/port: "9000"
>      38       {{- end }}
>      39     spec:
>      40       serviceAccountName: {{ template "graylog.serviceAccountName" . }}
>      41 {{- if .Values.graylog.nodeSelector }}
>      42       nodeSelector:
>      (...)

BR,
Catarina Tavares

**Comment by rafaelsilvaa10** [comment]
I have the same problem.

**Comment by rafaelsilvaa10** [comment]
Hello, I'm still the error.

➜  repos git clone https://github.com/helm/charts.git
Cloning into 'charts'...
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 67705 (delta 4), reused 7 (delta 2), pack-reused 67691
Receiving objects: 100% (67705/67705), 19.37 MiB | 3.19 MiB/s, done.
Resolving deltas: 100% (48228/48228), done.
Checking out files: 100% (5078/5078), done.
➜  repos cd charts/stable/graylog
➜  graylog git:(master) helm install --namespace "graylog" -n "graylog" stable/graylog
Error: validation failed: error validating "": error validating data: ValidationError(StatefulSet.spec.template.spec): unknown field "updateStrategy" in io.k8s.api.core.v1.PodSpec

**Comment by tru-cstavares** [comment]
Hi @rafaelsilvaa10,

Seems that helm is still getting chart 1.0.0, but the fix is on chart version 1.1.1.

Please add "--version 1.1.1" on command line as follows. I tested and deploy worked. 

```
helm  --debug install --namespace "graylog" -n "graylog" stable/graylog --version 1.1.1\
  --set tags.install-mongodb=false \
  --set tags.install-elasticsearch=false \
  --set graylog.mongodb.uri="mongodb://mongodb-mongodb-replicaset-0.mongodb-mongodb-replicaset.graylog.svc.cluster.local:27017/graylog?replicaSet=rs0" \
  --set graylog.elasticsearch.hosts=http://elasticsearch-client.graylog.svc.cluster.local:9200
```

BR.
